core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -172,15 +175,46 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
+  def getBillingProject(projectName: RawlsBillingProjectName): Future[PerRequestMessage] = {
+    for {
+      resourceIdsWithPolicyNames <- samDAO.getPoliciesForType(SamResourceTypeNames.billingProject, userInfo)
+      maybeBillingProject <- dataSource.inTransaction { dataAccess => dataAccess.rawlsBillingProjectQuery.load(projectName) }
+    } yield {
+      val bestRole = projectPoliciesToRoles(resourceIdsWithPolicyNames).foldLeft(Option.empty[ProjectRoles.ProjectRole]) {
```

Comment:
I think I might be misunderstanding this, but is this checking for the best role across all policies on all billing projects the user has access to?

Reply:
yes